lib/swagger/blocks.rb

```diff
@@ -347,55 +603,174 @@ def as_json
 
       def scope(&block)
         self.data[:_scopes] ||= []
-        self.data[:_scopes] << Swagger::Blocks::ApiAuthorizationScopeNode.call(&block)
+        self.data[:_scopes] << Swagger::Blocks::ApiAuthorizationScopeNode.call(version: version, &block)
       end
     end
 
-    # NOTE: in the spec this is different than Resource Listing's scope object.
-    # http://goo.gl/PvwUXj#5211-scope-object
+    # v1.2: NOTE: in the spec this is different than Resource Listing's scope object.
+    # v1.2: http://goo.gl/PvwUXj#5211-scope-object
     class ApiAuthorizationScopeNode < Node; end
 
-    # http://goo.gl/PvwUXj#434-items-object
+    # v2.0: https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#responseObject
+    class ResponseNode < Node
+      def schema(&block)
+        self.data[:schema] = Swagger::Blocks::SchemaNode.call(version: version, &block)
+      end
+
+      def header(head, &block)
+        # TODO validate 'head' is as per spec
+        self.data[:headers] ||= {}
+        self.data[:headers][head] = Swagger::Blocks::HeaderNode.call(version: version, &block)
+      end
+
+      def example(exam, &block)
+        # TODO validate 'exam' is as per spec
+        self.data[:examples] ||= {}
+        self.data[:examples][exam] = Swagger::Blocks::ExampleNode.call(version: version, &block)
+      end
+    end
+
+    class AllOfNode < Node
```

Comment:
Yeah, and this duplicates much of the as_json method from above as well. I'm fine with this for now since it's well-tested, but if this behavior is needed again we should rip it out.